File: 110-Balanced_Binary_Tree/main.js
var isBalanced = function(root) {
  if (root === null) {
    return true;
  }

  if (!isBalanced(root.left) || !isBalanced(root.right)) {
    return false;
  }

  return Math.abs(getHeight(root.left) - getHeight(root.right)) <= 1;
};

function getHeight(node) {
  if (node === null) {
    return 0;
  }

  const leH = getHeight(node.left);
  const riH = getHeight(node.right);
  return Math.max(leH, riH) + 1;
}

File: 110-Balanced_Binary_Tree/main.go
package bst

import (
	"math"
)

type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}

func isBalanced(root *TreeNode) bool {
	if root == nil {
		return true
	}

	if !isBalanced(root.Left) || !isBalanced(root.Right) {
		return false
	}

	return math.Abs(float64(getHeight(root.Left) - getHeight(root.Right))) <= 1
}

func getHeight(node *TreeNode) int {
	if node == nil {
		return 0
	}

	left := getHeight(node.Left)
	right := getHeight(node.Right)
	
	if left > right {
		return left + 1
	}
	return right + 1
}